### ISSUE TITLE
Regex DoS in HtmlImageSourceParser.EnsureImageSources

### ISSUE THREAD

**Issue by ocdi** [umbraco/Umbraco-CMS#8608]
A content editor was creating some content and copy/pasted some html from a different CMS. This contained some embedded data-uri images. At some point the content was attempted to be retrieved and the method EnsureImageSources is attempting to parse the contents, and causes it to use 100% CPU of a core. Multiple attempts to load and all cores are at 100%.

## Umbraco version

I am seeing this issue on Umbraco version: 8.6.3

Reproduction
------------
### Specifics

The regex here is copied from https://github.com/umbraco/Umbraco-CMS/blob/v8/8.7/src/Umbraco.Web/Templates/HtmlImageSourceParser.cs#L59

Note I added a timeout here.

```c#
        static readonly Regex ResolveImgPattern = new Regex(@"(<img[^>]*src="")([^""\?]*)([^""]*""[^>]*data-udi="")([^""]*)(""[^>]*>)",
                    RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace, TimeSpan.FromSeconds(5));

        const string TestContent = "<p>Y&amp;S this is content</p>\n<p><img style=\"display: block; margin-left: auto; margin-right: auto;\" src=\"data:image/png;base64,largedatahere=\" alt=\"\" /></p>";

        [Fact]
        public void TestRegexContent()
        {
            var result = ResolveImgPattern.Match(TestContent);
            Assert.False(result.Success);
        }
```

### Steps to reproduce

Create a content item with html content including large data-uri image.

### Expected result

The system to operate normally

### Actual result

100% of a core is used to evaluate the regex, never finishing.

**Comment by ocdi** [comment]
Given the test as it stands above actually passes, I thought I should provide a proper test case to demonstrate this flaw.

https://gist.github.com/ocdi/1c8cadabbccc9c40c83c6eba4b72e6e6

This data uri is only 64kb of data and this does succeed within 60 seconds, but this is way too slow for something that should be instant. I suspect it is exponentially slower for larger data-uri's.

I did an extremely large data URI which is 291KB in size and will leave this one running, burning CPU cycles until either xunit kills the test or I kill it.

https://gist.github.com/ocdi/50fdef866d4c6f754cb729cbe5ea39b5

**Comment by ocdi** [comment]
OK so it turns out I was wrong, it did finish evaluating after `19.2793 Minutes` on my workstation.

**Comment by ocdi** [comment]
This issue is a dupe of #7386 which has been solved already in the 8.7 branch, which is great to see. Just need the release to happen now. 👍
